components/performance/editor/PerformanceEditor.vue

```diff
@@ -1,47 +1,215 @@
+              @click="addSeatGroup()"
+            />
+          </div>
+          <div class="py-3 space-y-2">
+            <seat-group
+              v-for="(performanceSeatGroup, index) in performanceSeatGroups"
+              :key="performanceSeatGroup.id"
+              :value="performanceSeatGroup.seatGroup"
+              :removable="true"
+              @remove="performanceSeatGroups.splice(index, 1)"
+            />
+          </div>
+        </div>
+        <div class="px-2 border border-sta-gray rounded-lg">
+          <div class="flex items-center justify-between pt-3">
+            <h4 class="text-h4">Concessions</h4>
+            <font-awesome-icon
+              icon="plus-circle"
+              class="cursor-pointer text-lg hover:text-gray-300"
+              @click="addNewConcession()"
+            />
+          </div>
+          <div class="py-3 space-y-2">
+            <concession-type
+              v-for="discount in singleDiscounts"
+              :key="discount.id"
+              v-bind.sync="discount.requirements[0].concessionType"
+              :removable="true"
+              :editable="true"
+              @remove="deleteConcession(discount)"
+            />
+          </div>
+        </div>
+      </div>
+
+      <h4 class="mt-6 text-h4">Ticket Pricing</h4>
+      <div
+        v-if="!performanceSeatGroups.length"
+        class="p-4 text-white bg-sta-rouge"
+      >
+        Please add at least one seat group
+      </div>
+      <div
+        v-else-if="!singleDiscounts.length"
+        class="p-4 text-white bg-sta-rouge"
+      >
+        Please add at least one concession type
+      </div>
+      <div v-else class="max-w-full overflow-x-auto">
+        <div
+          v-if="!singleDiscounts.some((discount) => discount.percentage == 0)"
+          class="p-4 text-white bg-sta-rouge"
+        >
+          You need at least one concession type with 0% discount.
+        </div>
+        <table class="w-full">
+          <tr>
+            <th class="font-normal relative text-xs lg:text-sm">
+              <div
+                class="absolute diagonalCross p-2 pt-4 top-0 min-h-full w-full"
+              />
+              <div
+                class="absolute top-4 xl:top-5 right-2 w-20 lg:w-40 text-right"
+              >
+                Percentage Discount
+              </div>
+              <div class="absolute left-2 bottom-2 w-20 lg:w-40 text-left">
+                Seat Group Price
+              </div>
+            </th>
+            <th
+              v-for="discount in singleDiscounts"
+              :key="discount.id"
+              class="pb-2"
+            >
+              {{ discount.requirements[0].concessionType.name }}
+              <form-label>
+                <percentage-input
+                  :key="discount.id"
+                  :value="discount.percentage * 100"
+                  @input="discount.percentage = $event / 100"
+                />
+              </form-label>
+            </th>
```

Comment:
an issue exists where on changing the discount% of a concType, if the order of the concTypes changes (when sorted by discount% as they are) they swap in the editor and typing focus is lost - very annoying.
Observe:

The aim is to make conc B have a 25% discount 
![image](https://user-images.githubusercontent.com/61627133/147939524-91c882f4-2baf-4b32-ab75-37ef95bd061a.png)

On pressing the "5" key, the colums swap and focus is lost
![image](https://user-images.githubusercontent.com/61627133/147939555-cd984afe-bf9c-435e-875b-1273866d2908.png)

can the order be cahnged only after saving or something?